internal/provider/data_source_http_test.go

```diff
@@ -471,6 +472,36 @@ func TestDataSource_UnsupportedInsecureCaCert(t *testing.T) {
 	})
 }
 
+func TestDataSource_HTTPViaProxyWithEnv(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer svr.Close()
+
+	p := goproxy.NewProxyHttpServer()
+
+	proxy := httptest.NewServer(p)
+	defer proxy.Close()
+
+	t.Setenv("HTTP_PROXY", proxy.URL)
+	t.Setenv("HTTPS_PROXY", proxy.URL)
+
+	resource.UnitTest(t, resource.TestCase{
+		ProtoV5ProviderFactories: protoV5ProviderFactories(),
+
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+					data "http" "http_test" {
+						url = "%s"
+					}
+				`, svr.URL),
+				Check: resource.TestCheckResourceAttr("data.http.http_test", "status_code", "200"),
```

Comment:
This test looks correct to me, but for some reason is unhappy on Terraform version 0.14:

```
=== RUN   TestDataSource_HTTPViaProxyWithEnv
    data_source_http_test.go:489: Step 1/1 error: Check failed: Not found: data.http.http_test in [root]
--- FAIL: TestDataSource_HTTPViaProxyWithEnv (0.40s)
```

I'll try to do some investigative work with my end of day.

Reply:
The quirk with Terraform version 0.14.x is that it will not set the Terraform state for a data source which returns a _warning_ diagnostic. The warning being generated here is:

```
2022-11-01T16:50:56.310-0400 [TRACE] sdk.proto: Received downstream response: diagnostic_error_count=0 diagnostic_warning_count=1 tf_req_duration_ms=0 tf_data_source_type=http tf_req_id=5887d459-0e1a-5aa1-0241-13d21f8698ca tf_provider_addr=registry.terraform.io/hashicorp/http tf_proto_version=5.3 tf_rpc=ReadDataSource
2022-11-01T16:50:56.310-0400 [WARN]  sdk.proto: Response contains warning diagnostic: tf_provider_addr=registry.terraform.io/hashicorp/http tf_proto_version=5.3 tf_rpc=ReadDataSource tf_data_source_type=http diagnostic_severity=WARNING diagnostic_summary="Content-Type is not recognized as a text type, got \"\"" diagnostic_detail="If the content is binary data, Terraform may not properly handle the contents of the response." tf_req_id=5887d459-0e1a-5aa1-0241-13d21f8698ca
2022-11-01T16:50:56.310-0400 [TRACE] sdk.proto: Served request: tf_provider_addr=registry.terraform.io/hashicorp/http tf_proto_version=5.3 tf_rpc=ReadDataSource tf_data_source_type=http tf_req_id=5887d459-0e1a-5aa1-0241-13d21f8698ca
```

The data source correctly does not exit early in this case. The testing fix for this is to ensure the HTTP response also includes a valid `Content-Type` header, e.g.

```go
svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
	w.Header().Set("Content-Type", "text/plain")
	w.WriteHeader(http.StatusOK)
}))
```

Reply:
Thanks for the investigation. Have added the fix you suggested.